src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseXmlReaderForDataSetReadXmlTests.cs

```diff
@@ -0,0 +1,366 @@
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXml(new StreamReader(""TestFile.txt""), XmlReadMode.Auto);
+    }
+}",
+            GetCSharpResultAt(10, 9, UseXmlReaderForDataSetReadXml.RealRule, "DataSet", "ReadXml"));
+        }
+
+        [Fact]
+        public void TestReadXmlSchemaWithStreamParameterDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXmlSchema(new FileStream(""xmlFilename"", FileMode.Open));
+    }
+}",
+            GetCSharpResultAt(10, 9, UseXmlReaderForDataSetReadXml.RealRule, "DataSet", "ReadXmlSchema"));
+        }
+
+        [Fact]
+        public void TestReadXmlSchemaWithStringParameterDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXmlSchema(""Filename"");
+    }
+}",
+            GetCSharpResultAt(9, 9, UseXmlReaderForDataSetReadXml.RealRule, "DataSet", "ReadXmlSchema"));
+        }
+
+        [Fact]
+        public void TestReadXmlSchemaWithTextReaderParameterDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXmlSchema(new StreamReader(""TestFile.txt""));
+    }
+}",
+            GetCSharpResultAt(10, 9, UseXmlReaderForDataSetReadXml.RealRule, "DataSet", "ReadXmlSchema"));
+        }
+
+        [Fact]
+        public void TestReadXmlWithXmlReaderParameterNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+using System.Xml;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXml(new XmlTextReader(new FileStream(""xmlFilename"", FileMode.Open)));
+    }
+}");
+        }
+
+        [Fact]
+        public void TestReadXmlWithXmlReaderAndXmlReadModeParametersNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+using System.Xml;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXml(new XmlTextReader(new FileStream(""xmlFilename"", FileMode.Open)), XmlReadMode.Auto);
+    }
+}");
+        }
+
+        [Fact]
+        public void TestReadXmlSchemaWithXmlReaderParameterNoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Data;
+using System.IO;
+using System.Xml;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        new DataSet().ReadXmlSchema(new XmlTextReader(new FileStream(""xmlFilename"", FileMode.Open)));
+    }
+}");
+        }
+
+        [Fact]
+        public void TestReadXmlSerializableWithXmlReaderParameterNoDiagnostic()
+        {
```

Comment:
Are there any override test cases with a diagnostic?

Reply:
There's only one method ReadXmlSerializable with modifier virtual in DataSet and it won't produce diagnostic cause the type of its parameter is XmlReader which meets the requirment.

Reply:
But UseXmlReaderForDeserialize has tests with a diagnostic on override method.